src/linear_algebra/general_linear_group.lean

```diff
@@ -0,0 +1,167 @@
+  map_one' := units.ext det_one,
+  map_mul' := λ A B, units.ext $ det_mul _ _ }
+
+/--The `GL n R` and `general_linear_group R n` groups are multiplicatively equivalent-/
+def to_lin : (GL n R) ≃* (linear_map.general_linear_group R (n → R)) :=
+units.map_equiv to_lin_alg_equiv'.to_mul_equiv
+
+/--Given a matrix with invertible determinant we get an element of `GL n R`-/
+def mk' (A : matrix n n R) (h : invertible (matrix.det A)) : GL n R :=
+unit_of_det_invertible A
+
+/--Given a matrix with unit determinant we get an element of `GL n R`-/
+noncomputable def mk'' (A : matrix n n R) (h : is_unit (matrix.det A)) : GL n R :=
+nonsing_inv_unit A h
+
+instance coe_fun : has_coe_to_fun (GL n R) :=
+{ F   := λ _, n → n → R,
+  coe := λ A, A.val }
+
+lemma ext_iff (A B : GL n R) : A = B ↔ (∀ i j, (A : matrix n n R) i j = (B : matrix n n R) i j) :=
+units.ext_iff.trans matrix.ext_iff.symm
+
+/-- Not marked `@[ext]` as the `ext` tactic already solves this. -/
+lemma ext ⦃A B : GL n R⦄ (h : ∀ i j, (A : matrix n n R) i j = (B : matrix n n R) i j) :
+  A = B :=
+units.ext $ matrix.ext h
+
+section coe_lemmas
+
+variables (A B : GL n R)
+
+@[simp] lemma coe_fn_eq_coe : ⇑A = (↑A : matrix n n R) := rfl
+
+@[simp] lemma mul_val : ↑(A * B) = A ⬝ B := rfl
+
+@[simp] lemma one_val : ↑(1 : GL n R) = (1 : matrix n n R) := rfl
+
+@[simp] lemma coe_inv : ↑(A⁻¹) = (↑A : matrix n n R)⁻¹ :=
+begin
+  letI := A.invertible,
+  exact inv_eq_nonsing_inv_of_invertible (↑A : matrix n n R),
+end
+
+@[simp] lemma to_lin'_one : to_lin' (1 : GL n R).1 = linear_map.id :=
+matrix.to_lin'_one
+
+end coe_lemmas
+
+instance SL_to_GL : has_coe (special_linear_group n R) (GL n R) :=
+⟨λ A, ⟨↑A, ↑(A⁻¹), congr_arg coe (mul_right_inv A), congr_arg coe (mul_left_inv A)⟩⟩
+
+end general_linear_group
+
+namespace GL_plus
+
+variables {n : Type u} {R : Type v} [decidable_eq n] [fintype n] [linear_ordered_comm_ring R ]
+
+lemma one_in_GL_pos : 0 < det (1 : GL n R) :=
+begin
+simp only [det_one, general_linear_group.coe_fn_eq_coe, general_linear_group.one_val, zero_lt_one],
+end
+
+lemma mul_in_GL_pos (A B : GL n R) (h1 : 0 < det A) (h2 : 0 < det B) : 0 < det (A*B) :=
+begin
+  simp only [gt_iff_lt, det_mul, mul_eq_mul],
+  apply mul_pos h1 h2,
+end
+
+-- this is `general_linear_group.det` which is always a unit!
+lemma inv_det_pos (A : GL n R) (h : (0 : R) < A.det) : (0 : R) < A⁻¹.det :=
+begin
+  have : (0 : R) < (A * A⁻¹).det := by simp,
+  rw [monoid_hom.map_mul, units.coe_mul] at this,
+  exact (zero_lt_mul_left h).mp this,
+end
+
+section
+variables (n R)
+
+/-- This is the subgroup of `nxn` matrices with entries over a
+linear ordered ring and positive determinant. -/
+def GL_pos : subgroup (GL n R) :=
+{ carrier := {M  | 0 < det M},
+  one_mem' := one_in_GL_pos  ,
+  mul_mem' := λ A B h1 h2, mul_in_GL_pos  A B h1 h2,
+  inv_mem' := λ A h1, begin
+    simp only [set.mem_set_of_eq] at *,
+    have:= inv_det_pos  A h1,
+    unfold_coes at *,
+    convert this,
+  end }
```

Comment:
I've come up with a possibly nicer way to construct this; using the more general results from [this Zulip thread](https://leanprover.zulipchat.com/#narrow/stream/217875-Is-there.20code.20for.20X.3F/topic/The.20submonoid.20of.20positive.20elements/near/247798909), this becomes
```
def GL_pos : subgroup (GL n R) := units.pos_subgroup.comap general_linear_group.det
```

Reply:
Oh I think thats a great idea, may as well do it this way as this doesn't use anything specific about matrices. Where would the other definitions need to be added? would it be in the monoid section?

Reply:
I'm not sure yet where would be best to add them. For now, you could add them at the top of the file with a comment saying "TODO: move these", and ask another reviewer for help / wait for a response on zulip.

Reply:
Ok cool. I've done that, lets see if anyone replies on Zulip.